website/docs/reference/transforms/rename_fields.md.erb

```diff
@@ -32,7 +32,18 @@
 
 ### Key Conflicts
 
-Keys specified in this transform will replace existing keys.
+Keys specified in this transform will replace existing keys _in full_. For
+example, if a target key contains an object value it will be replaced _entirely_
+by the new key's value, even if the new value is an object itself. Vector will
+_not_ perform a deep merge.
```

Comment:
@Hoverbear is this true? Should we add a behavior test for this because I have a feeling this will change with @a-rodin nested field changes.

Reply:
They *should* provide a deep merge to my understanding of the feature. Do they not currently?

Here's my test config:

```toml
data_dir = "/var/lib/vector/"
dns_servers = []

[log_schema]
message_key = "message"
timestamp_key = "timestamp"
host_key = "host"

[sources.source0]
max_length = 102400
type = "stdin"

[transforms.transform0]
inputs = ["source0"]
type = "add_fields"
[transforms.transform0.fields]
    a.a = "a"
    a.b = "b"

[transforms.transform1]
inputs = ["transform0"]
type = "rename_fields"
[transforms.transform1.fields]
    a.a = "a.c" # example

[sinks.sink0]
healthcheck = true
inputs = ["transform1"]
type = "console"
encoding = "json"

[sinks.sink0.buffer]
type = "memory"
max_events = 500
when_full = "block"
```

Here's the output:

```bash
Feb 24 09:16:18.550  INFO vector: Log level "info" is enabled.
Feb 24 09:16:18.550  INFO vector: Loading configs. path=["rename.toml"]
Feb 24 09:16:18.565  INFO vector: Vector is starting. version="0.8.0" git_version="v0.7.0-199-g794e473" released="Mon, 24 Feb 2020 16:57:05 +0000" arch="x86_64"
Feb 24 09:16:18.567  INFO vector::topology: Running healthchecks.
Feb 24 09:16:18.567  INFO vector::topology: Starting source "source0"
Feb 24 09:16:18.568  INFO vector::topology::builder: Healthcheck: Passed.
Feb 24 09:16:18.568  INFO vector::topology: Starting transform "transform0"
Feb 24 09:16:18.568  INFO source{name=source0 type=stdin}: vector::sources::stdin: Capturing STDIN
Feb 24 09:16:18.569  INFO vector::topology: Starting transform "transform1"
Feb 24 09:16:18.569  INFO vector::topology: Starting sink "sink0"
asd
{"a":{"b":"b","c":"a"},"host":"OBSIDIAN","message":"asd","timestamp":"2020-02-24T17:16:19.513806924Z"}
a   
{"a":{"b":"b","c":"a"},"host":"OBSIDIAN","message":"a","timestamp":"2020-02-24T17:16:32.182928767Z"}
```

Reply:
Ok, how will we preserve this behavior when #1902 and #1662 land?

Reply:
Are those PRs changing what is expected here?

Can you confirm what is expected when a logevent is:

```json
{ "something": 1, "parent": { "child": 2, "other": 3 } }
```

And we put it through a rename that renames `something` to `parent.something`?

Is it A:

```json
{ "parent": { "child": 2, "other": 3, "something": 1 } }
```

or B:

```json
{ "parent": {  "something": 1 } }
```

Reply:
That’s up to you, but it should be documented and tested so we don’t regress. If I had to pick I’d say A in your example, but that’s also not the example this come this referencing.

Reply:
I would just think through object and value merging and go with what you feel is best. I just want to make sure we don’t change behavior when we get the new nested field changes in place.

Reply:
Ok. :) The behavior of replacing an existing field is tested:

https://github.com/timberio/vector/blob/d3106e4252d6226925965e001940f8ef44bb4289/tests/behavior/transforms/rename_fields.toml#L7-L8

https://github.com/timberio/vector/blob/d3106e4252d6226925965e001940f8ef44bb4289/tests/behavior/transforms/rename_fields.toml#L35-L36